tests/test_ops.py

```diff
@@ -1,173 +1,200 @@
+        ]
+    )
+    assert np.allclose(matrix, true_matrix)
+
+
+@pytest.mark.parametrize("phi", phis)
+@pytest.mark.parametrize("mag", mags)
+def test_two_mode_squeezing_heisenberg(phi, mag):
+    """ops: Tests the Heisenberg representation of the Beamsplitter gate."""
+    r = mag
+    matrix = cv.TwoModeSqueezing._heisenberg_rep([r, phi])
+    true_matrix = np.array(
+        [
+            [1, 0, 0, 0, 0],
+            [0, np.cosh(r), 0, np.cos(phi) * np.sinh(r), np.sin(phi) * np.sinh(r)],
+            [0, 0, np.cosh(r), np.sin(phi) * np.sinh(r), -np.cos(phi) * np.sinh(r)],
+            [0, np.cos(phi) * np.sinh(r), np.sin(phi) * np.sinh(r), np.cosh(r), 0],
+            [0, np.sin(phi) * np.sinh(r), -np.cos(phi) * np.sinh(r), 0, np.cosh(r)],
+        ]
+    )
+    assert np.allclose(matrix, true_matrix)
+
+
+@pytest.mark.parametrize("s", s_vals)
+def test_quadratic_phase_heisenberg(s):
+    """ops: Tests the Heisenberg representation of the QuadraticPhase gate."""
+    matrix = cv.QuadraticPhase._heisenberg_rep([s])
+    true_matrix = np.array([[1, 0, 0], [0, 1, 0], [0, s, 1]])
+    assert np.allclose(matrix, true_matrix)
+
+
+@pytest.mark.parametrize("s", s_vals)
+def test_controlled_addition_heisenberg(s):
+    """ops: Tests the Heisenberg representation of ControlledAddition gate.
+    """
+    matrix = cv.ControlledAddition._heisenberg_rep([s])
+    true_matrix = np.array(
+        [
+            [1, 0, 0, 0, 0],
+            [0, 1, 0, 0, 0],
+            [0, 0, 1, 0, -s],
+            [0, s, 0, 1, 0],
+            [0, 0, 0, 0, 1],
+        ]
+    )
+    assert np.allclose(matrix, true_matrix)
+
+
+@pytest.mark.parametrize("s", s_vals)
+def test_controlled_phase_heisenberg(s):
+    """Tests the Heisenberg representation of the ControlledPhase gate."""
+    matrix = cv.ControlledPhase._heisenberg_rep([s])
+    true_matrix = np.array(
+        [
+            [1, 0, 0, 0, 0],
+            [0, 1, 0, 0, 0],
+            [0, 0, 1, s, 0],
+            [0, 0, 0, 1, 0],
+            [0, s, 0, 0, 1],
+        ]
+    )
+    assert np.allclose(matrix, true_matrix)
+
+
+class TestNonGaussian:
+    """Tests that non-Gaussian gates are properly handled."""
+
+    @pytest.mark.parametrize("gate", [cv.Kerr, cv.CrossKerr, cv.CubicPhase])
+    def test_heisenberg_rep_nonguassian(self, gate):
+        """ops: Tests that the `_heisenberg_rep` for a non-Gaussian gates is
+        None
+        """
+        assert gate._heisenberg_rep(*[0.1] * gate.num_params) is None
+
+    def test_heisenberg_transformation_nongaussian(self):
+        """ops: Tests that proper exceptions are raised if we try to call the
+        Heisenberg transformation of non-Gaussian gates."""
+        op = cv.Kerr
+        with pytest.raises(RuntimeError, match=r"not a Gaussian operation"):
+            op_ = op(*[0.1] * op.num_params, [0] * op.num_wires, do_queue=False)
+            op_.heisenberg_tr(op.num_wires)
+
+        op = cv.CrossKerr
+        with pytest.raises(ValueError):
```

Comment:
Just a reminder for us to make an issue on this `ValueError`, seems confusing to not have this match the others.